### PR TITLE
print: unique name at print, add print database, add print limiter

### PR DESCRIPTION
**Pull request by andi34**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/PhotoboothProject/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)

- [ ] Documentation update
- [x] Bug fix
- [x] New feature
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)
While changing printing settings it might be necessary to recreate the image for print. Instead we were using an old resource with old config applied.
Image for print now will be saved with a unique name, keeping the old name in front to use for statistics if necessary.

Log printed images inside a database (csv) especially to allow counting made prints.

Allow to limit prints:
- print get locked every defined amount of prints, 0 for unlimited prints allowed
- Define a message which will be visible while print gets locked,
- Define a message which is visible once print is locked.

Currently print can only be unlocked from the adminpanel, adding other options to unlock can be discussed.

### CODE REVIEW

**Comment by andi34** [comment]
Allow to disable Print database via config?

**Comment by andi34** [comment]
Looks good to me. Wouldn't like it make it optional as it won't hurt.

Suggestions before merge?
Adding a print limit can be considered, but should go into a different PR.

File: api/admin.php [hunk 1]
```diff
@@ -194,6 +194,14 @@
                 }
             }
 
+            if ($newConfig['reset']['remove_print_db']) {
+                // delete print log
```
Comment (andi34) [inline api/admin.php]: DB

**Comment by andi34** [comment]
settings

locking

locked

locked-1

**Comment by MatthiasWit** [comment]
Is it Possible to show the number of printed on the home page?

**Pull request merged**